Prevent segfaults when using simulatormodule without a simulator (#1843)
cocotb/cocotb: documentation/source/newsfragments/1843.change.rst

> Simulator Python module now throws an exception if used when no simulator is present. Making the use of cocotb without a simulator present safe.

Reviewer: ```suggestion
The :mod:`cocotb.simulator` Python module now throws an exception if used when no simulator is present, making the use of cocotb without a simulator present safe.
```
(Cannot test right now if this will result in the correct link.)

> The :mod:`cocotb.simulator` Python module now throws an exception if used when no simulator is present, making the use of cocotb without a simulator present safe.

Reviewer 2: ```suggestion
The :mod:`cocotb.simulator` Python module now throws an :exc`RuntimeError` if used when no simulator is present, making it safe to use cocotb without a simulator present.
```
